### PR TITLE
Make TPM 1.2 optional, add --with-tpm1 configure option

### PR DESCRIPTION
**Pull request by elmarco**

### CODE REVIEW

File: src/tpm_library.c [hunk 1]
```diff
@@ -110,11 +116,13 @@ TPM_RESULT TPMLIB_ChooseTPMVersion(TPMLIB_TPMVersion ver)
 
     switch (ver) {
     case TPMLIB_TPM_VERSION_1_2:
+#if WITH_TPM1
         if (tpmvers_choice != 0)
             ClearAllCachedState();
 
         tpmvers_choice = 0; // entry 0 in tpm_iface
         break;
+#endif
```
[the remark this answers is not in the document]
Comment (elmarco) [reply src/tpm_library.c:125]: oops, no. I copy the TPM2 case, but that's not good enough!

File: configure.ac [hunk 2]
```diff
@@ -67,6 +67,10 @@ m4_if(
 
 AM_CONDITIONAL([HAVE_VERSION_SCRIPT], [test "x$have_version_script" = "xyes"])
 
+AC_ARG_WITH([tpm1],
+    AS_HELP_STRING([--with-tpm1], [build libtpms with TPM1 support]), [], [with_tpm1=yes])
+AM_CONDITIONAL([WITH_TPM1], [test "x$with_tpm1" != "xno"])
+AS_IF([test "x$with_tpm1" != "xno"], [AC_DEFINE([WITH_TPM1], [1], [With TPM 1.2 support])])
```
Comment (stefanberger) [inline configure.ac:73]: I wonder whether there should be a variable like this in libtpms.pc as well so that swtpm knows it cannot run the TPM 1.2 tests.

Comment (elmarco) [reply configure.ac:73]: Sure, that could help to avoid run time skip.

Comment (elmarco) [reply configure.ac]: done

File: configure.ac [hunk 3]
```diff
@@ -80,6 +83,7 @@ AC_ARG_WITH([tpm2],
 AS_IF([test "x$with_tpm2" = xyes], [
 	AC_MSG_RESULT([Building with TPM2 support])
 	AC_DEFINE_UNQUOTED([WITH_TPM2], 1, [whether to support TPM2])
+    AC_SUBST([WITH_TPM2], [1])
```
Comment (stefanberger) [inline configure.ac:86]: indentation

File: configure.ac [hunk 4]
```diff
@@ -67,6 +67,10 @@ m4_if(
 
 AM_CONDITIONAL([HAVE_VERSION_SCRIPT], [test "x$have_version_script" = "xyes"])
 
+AC_ARG_WITH([tpm1],
+    AS_HELP_STRING([--with-tpm1], [build libtpms with TPM1 support]), [], [with_tpm1=yes])
```
Comment (stefanberger) [inline configure.ac:71]: to be consistent with the text below :... with TPM1.2 support

File: src/disabled_interface.c [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    return TPM_FAIL;
+}
+
+static void Disabled_Terminate(void)
+{
+}
+
+static TPM_RESULT Disabled_Process(unsigned char **respbuffer, uint32_t *resp_size,
+                               uint32_t *respbufsize,
+                               unsigned char *command, uint32_t command_size)
```
Comment (stefanberger) [inline src/disabled_interface.c:16]: Please indent and all the ones below as well.

**Comment by stefanberger** [comment]
Thanks!

**Pull request merged**